sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryClient.java

```diff
@@ -224,126 +199,104 @@ public DirectoryProperties getProperties() {
-        return directoryAsyncClient.setMetadataWithResponse(metadata, context).block();
+    public Response<DirectorySetMetadataInfo> setMetadataWithResponse(Map<String, String> metadata, Duration timeout, Context context) {
+        Mono<Response<DirectorySetMetadataInfo>> response = directoryAsyncClient.setMetadataWithResponse(metadata, context);
+        return Utility.blockWithOptionalTimeout(response, timeout);
     }
 
     /**
      * Lists all sub-directories and files in this directory without their prefix or maxResult.
-     *
      * <p><strong>Code Samples</strong></p>
-     *
      * <p>List all sub-directories and files in the account</p>
-     *
      * {@codesnippet com.azure.storage.file.directoryClient.listFilesAndDirectories}
-     *
      * <p>For more information, see the
      * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/list-directories-and-files">Azure Docs</a>.</p>
      *
      * @return {@link FileRef File info} in the storage directory
      */
     public PagedIterable<FileRef> listFilesAndDirectories() {
-        return listFilesAndDirectories(null, null);
+        return listFilesAndDirectories(null, null, null);
     }
 
     /**
      * Lists all sub-directories and files in this directory with their prefix or snapshots.
-     *
      * <p><strong>Code Samples</strong></p>
-     *
      * <p>List all sub-directories and files in this directory with "subdir" prefix and return 10 results in the account</p>
-     *
-     * {@codesnippet com.azure.storage.file.directoryClient.listFilesAndDirectories#string-integer}
-     *
+     * {@codesnippet com.azure.storage.file.directoryClient.listFilesAndDirectories#string-integer-duration}
      * <p>For more information, see the
      * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/list-directories-and-files">Azure Docs</a>.</p>
      *
      * @param prefix Optional prefix which filters the results to return only files and directories whose name begins with.
      * @param maxResults Optional maximum number of files and/or directories to return per page.
-     *                   If the request does not specify maxresults or specifies a value greater than 5,000, the server will return up to 5,000 items.
+     * If the request does not specify maxresults or specifies a value greater than 5,000, the server will return up to 5,000 items.
+     * @param timeout An optional timeout value beyond which a {@link RuntimeException} will be raised.
      * @return {@link FileRef File info} in this directory with prefix and max number of return results.
      */
-    public PagedIterable<FileRef> listFilesAndDirectories(String prefix, Integer maxResults) {
-        return new PagedIterable<>(directoryAsyncClient.listFilesAndDirectories(prefix, maxResults));
+    public PagedIterable<FileRef> listFilesAndDirectories(String prefix, Integer maxResults, Duration timeout) {
```

Comment:
Looks like the listing methods are missing the Context parameter in the sync class, mind adding that as well?